core-test/src/test/scala/org/bitcoins/core/psbt/PSBTTest.scala

```diff
@@ -136,6 +136,48 @@ class PSBTTest extends BitcoinSAsyncTest {
     assert(psbt2.combinePSBT(psbt2) != expected)
   }
 
+  it must "correctly combine PSBTs" in {
+    forAllAsync(PSBTGenerators.psbtWithUnknowns) { psbtF =>
+      psbtF.map { psbt =>
+        val global = psbt.globalMap.elements
+        val inputs = psbt.inputMaps
+        val outputs = psbt.outputMaps
+
+        val newGlobalElements = PSBTGenerators.pruneVec(global) :+ GlobalPSBTRecord
+          .UnsignedTransaction(psbt.transaction)
+        val newGlobal = GlobalPSBTMap(newGlobalElements.distinct)
```

Comment:
This seems like a common thing to do, we should add a util next to `pruneVec` called `def pruneGlobal(globalMap: GlobalPSBTMap): GlobalPSBTMap`